Headers/Core/Engine/Map.h

```diff
@@ -17,12 +17,12 @@ class Map : public Node {
     Map();
     ~Map();
     
-    void addTile( Tile* tile );
+    void addTile( std::shared_ptr<Tile> tile );
     void setup();
     
     int rows;
     int columns;
-    std::vector<Tile*> tiles;
+    std::vector<std::shared_ptr<Tile>> tiles;
```

Comment:
Are the tiles being shared with some other object? In other case this kind of pointers should be unique_ptr or raw.